e2e/cli/context_test.go

```diff
@@ -108,4 +109,57 @@ var _ = Describe("home context", func() {
 			Expect(contexts.Current).To(Equal(defaultContext))
 		})
 	})
+
+	Describe("add a podman context", Ordered, func() {
+		testContext := "envd_home_test"
+		testBuilder := types.BuilderTypePodman
+		testBuilderAddress := "envd_buildkitd"
+		testRunner := types.RunnerTypePodman
+		c := types.Context{
+			Name:           testContext,
+			Builder:        testBuilder,
+			BuilderAddress: testBuilderAddress,
+			Runner:         testRunner,
+		}
+		originalEnv := os.Getenv("DOCKER_HOST")
+		BeforeAll(func() {
+			os.Setenv("DOCKER_HOST", "unix:///run/user/1000/podman/podman.sock")
+			err := home.GetManager().ContextCreate(c, true)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("should find a new context", func() {
+			contexts, err := home.GetManager().ContextList()
+			Expect(err).NotTo(HaveOccurred())
+			Expect(contexts.Current).To(Equal(testContext))
+		})
+
+		Describe("connect buildkit through podman-container", Ordered, func() {
+			buildContext := "testdata/build-test"
+
+			It("should be able to build image with podman context", func() {
+				args := []string{"envd.test", "--debug", "build", "--path", buildContext}
```

Comment:
envd build failed on my local test, no clue for now.
```shell
time="2022-11-02T13:15:39+08:00" level=error msg="Buildkit error: failed to receive status: rpc error: code = Unavailable desc = error reading from server: command [podman exec -i envd_buildkitd buildctl dial-stdio] has exited with exit status 255, please make sure the URL is valid, and Docker 18.09 or later is installed on the remote host: stderr=Error: timed out waiting for file /home/taylor/.local/share/containers/storage/overlay-containers/784926ab53d9d7ae579d992f36a361ca7c6ec89465b50919e197583876bd1fe8/userdata/1fbb00f5e2a6dc22afbeac7a4d6cea2dd83c0dfabb7a665107a77f54142175a6/exit/784926ab53d9d7ae579d992f36a361ca7c6ec89465b50919e197583876bd1fe8: internal libpod error\n(1) attached stack trace\n  -- stack trace:\n  | github.com/tensorchord/envd/pkg/builder.generalBuilder.build.func1\n  | \t/home/taylor/src/github.com/TaylorHere/envd/pkg/builder/builder.go:296\n  | golang.org/x/sync/errgroup.(*Group).Go.func1\n  | \t/home/taylor/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:57\n  | runtime.goexit\n  | \t/usr/lib/go-1.18/src/runtime/asm_amd64.s:1571\nWraps: (2) Buildkit error\nWraps: (3) failed to receive status: rpc error: code = Unavailable desc = error reading from server: command [podman exec -i envd_buildkitd buildctl dial-stdio] has exited with exit status 255, please make sure the URL is valid, and Docker 18.09 or later is installed on the remote host: stderr=Error: timed out waiting for file /home/taylor/.local/share/containers/storage/overlay-containers/784926ab53d9d7ae579d992f36a361ca7c6ec89465b50919e197583876bd1fe8/userdata/1fbb00f5e2a6dc22afbeac7a4d6cea2dd83c0dfabb7a665107a77f54142175a6/exit/784926ab53d9d7ae579d992f36a361ca7c6ec89465b50919e197583876bd1fe8: internal libpod error\n  | (1) failed to receive status: rpc error: code = Unavailable desc = error reading from server: command [podman exec -i envd_buildkitd buildctl dial-stdio] has exited with exit status 255, please make sure the URL is valid, and Docker 18.09 or later is installed on the remote host: stderr=Error: timed out waiting for file /home/taylor/.local/share/containers/storage/overlay-containers/784926ab53d9d7ae579d992f36a361ca7c6ec89465b50919e197583876bd1fe8/userdata/1fbb00f5e2a6dc22afbeac7a4d6cea2dd83c0dfabb7a665107a77f54142175a6/exit/784926ab53d9d7ae579d992f36a361ca7c6ec89465b50919e197583876bd1fe8: internal libpod error\n  | Error types: (1) *builder.BuildkitdErr\nError types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *builder.BuildkitdErr"
```

Reply:
```shell
time="2022-11-02T13:15:39+08:00" level=error msg="failed to load docker image: Post \"http://%2Frun%2Fuser%2F1000%2Fpodman%2Fpodman.sock/v1.40/images/load?quiet=1\": context canceled" tag="build-test:dev"
```

Reply:
Maybe you could run `podman exec -i envd_buildkitd buildctl dial-stdio` to get the error log.

Reply:
the logs from `podman --log-level=debug system service`
```shell
DEBU[0017] Loading image from "/var/tmp/api_load.tar821975805" 
DEBU[0017] -> Attempting to load "/var/tmp/api_load.tar821975805" as an OCI directory 
DEBU[0017] parsed reference into "[overlay@/home/taylor/.local/share/containers/storage+/run/user/1000/containers]localhost/var/tmp/api_load.tar821975805:latest" 
DEBU[0017] Copying source image /var/tmp/api_load.tar821975805: to destination image [overlay@/home/taylor/.local/share/containers/storage+/run/user/1000/containers]localhost/var/tmp/api_load.tar821975805:latest 
DEBU[0017] Error loading /var/tmp/api_load.tar821975805: initializing source oci:/var/tmp/api_load.tar821975805:: open /var/tmp/api_load.tar821975805/index.json: not a directory 
DEBU[0017] -> Attempting to load "/var/tmp/api_load.tar821975805" as an OCI archive 
DEBU[0017] Error loading /var/tmp/api_load.tar821975805: creating temp directory: untarring file "/var/tmp/oci267797681": unexpected EOF 
DEBU[0017] -> Attempting to load "/var/tmp/api_load.tar821975805" as a Docker dir 
DEBU[0017] parsed reference into "[overlay@/home/taylor/.local/share/containers/storage+/run/user/1000/containers]localhost/var/tmp/api_load.tar821975805:latest" 
DEBU[0017] Copying source image /var/tmp/api_load.tar821975805 to destination image [overlay@/home/taylor/.local/share/containers/storage+/run/user/1000/containers]localhost/var/tmp/api_load.tar821975805:latest 
DEBU[0017] Using blob info cache at /home/taylor/.local/share/containers/cache/blob-info-cache-v1.boltdb 
DEBU[0017] Error loading /var/tmp/api_load.tar821975805: determining manifest MIME type for dir:/var/tmp/api_load.tar821975805: open /var/tmp/api_load.tar821975805/manifest.json: not a directory 
DEBU[0017] -> Attempting to load "/var/tmp/api_load.tar821975805" as a Docker archive 
DEBU[0017] No compression detected                      
DEBU[0017] Error loading /var/tmp/api_load.tar821975805: loading tar component manifest.json: unexpected EOF 
DEBU[0017] Error loading /var/tmp/api_load.tar821975805: payload does not match any of the supported image formats (oci, oci-archive, dir, docker-archive) 
INFO[0017] Request Failed(Internal Server Error): failed to load image: payload does not match any of the supported image formats (oci, oci-archive, dir, docker-archive) 
@ - - [02/Nov/2022:22:17:31 +0800] "POST /v1.40/images/load?quiet=1 HTTP/1.1" 500 257 "" "Go-http-client/1.1"
```

Reply:
Interesting. Maybe we should open an issue for it.